### ISSUE TITLE
hs.eventtap mouse events stopped working in 0.9.56?

### ISSUE THREAD

**Issue by jaens** [Hammerspoon/hammerspoon#1547]
I have the following code which completely stopped working after an upgrade from 0.9.55 to 0.9.56:
(nothing is printed)
```lua
local ev = hs.eventtap.event

tap = hs.eventtap.new(
    {ev.types.middleMouseDown, ev.types.middleMouseUp,
    ev.types.otherMouseDown, ev.types.otherMouseUp},
    function(e)
        hs.printf('enter')
        --- other code
    end
)
tap:start()
```

**Comment by cmsj** [comment]
Hilariously, I only have my laptop with me this week, so I can't easily generate middle mouse button events, but I tested this for leftMouseDown/leftMouseUp and it worked fine.

@jaens could you confirm that the code works with left mouse events?

@asmagill it's suspicious that the middle mouse button is rearing its head again - are you able to reproduce this?

**Comment by asmagill** [comment]
If that worked after `middleMouseUp` and `middleMouseDown` were re-added to Hammerspoon (i.e. somewhere in between this release and the last) then I'm really surprised...

Eventtap works by building a mask from the events you list in the table by exclusively-or'ing them together -- this was done to support the use of the string "all" as a catchall for experimenting and trying to identify newly added events (see the docs for `hs.eventtap.new`, though looking at it I should perhaps make it more clear).

What this means is that the above code is sending the table `{ 25, 26, 25, 26 }` which is converted to a mask by doing the following:

1. toggle bit 25 (on)
2. toggle bit 26 (on)
3. toggle bit 25 (off)
4. toggle bit 26 (off)

 -- basically "watch for no events".

List only one set of "middleMouseDown, middleMouseUp" or "otherMouseDown, otherMouseUp", not both and it should work as expected.

**Comment by jaens** [comment]
Alright, thanks for the explanation.
Seems like this means that if any event type is listed in the table twice (or an even number of times), it will actually be turned off? Seems potentially confusing when building the event set by eg. concatenating tables :)

**Comment by asmagill** [comment]
In essence, yes, listing the same event an even number of times is the same as not listing it at all.

I'm going to leave this open so I can remember to update the documentation to make it more clear.
